selftests/pre_release/tests/vmimage.py

```diff
@@ -69,4 +71,6 @@ class ImageFunctional(Base):
     def test_get(self):
         cmd = 'avocado vmimage get --distro=%s --distro-version=%s --arch=%s'
         cmd %= (self.vmimage_name, self.vmimage_version, self.vmimage_arch)
-        process.run(cmd)
+        out = process.run(cmd)
+        image_path = out.stdout_text.split()[-1]
+        shutil.rmtree(os.path.dirname(image_path))
```

Comment:
This is nice, but shouldn't this be part of teardown method?

Reply:
I was thinking about it, but I think that with teardown method it will be more complicated to get the path to the just downloaded image and I don't see a benefit of teardown method in this case. So I used an easy solution.

Reply:
AFAICT, we are exporting to our users both: setup() and tearDown() methods. Those are facilities that are executed at the beginning and end of tests. Even when something fails inside the test.

For instance, if your process.run() fails to run at some level and/or your split()[-1] fails, shutil will not be executed. That would not happen if it was at tearDown().

IIUC, you can use setup() to define an object property that could be accessed at tearDown(). Unless I'm missing something, this could be possible, for instance:

```python
def setup(self, arg):
  self.image_path = None
   return super().setup(arg)

def test_foo():
   self.image_path = something

def tearDown():
   shutil.rmtree(os.path.dirname(image_path)) 

``` 

Like as said, unless I'm missing something, I would consider this approach. Otherwise, we would be "selling" something and using another approach. And I would vote to change this.

Reply:
> For instance, if your process.run() fails to run at some level and/or your split()[-1] fails, shutil will not be executed. That would not happen if it was at tearDown().

Actually this behavior was my intention. Because if there will be some fail in the test, there is no way how can I get the `image_path` (maybe scan avocado cache for changes) so I don't want to execute shutil. That is why I don't see a benefit of teardown.

Reply:
Not sure if I understood. If you download the file on line 74 and in the future we improve this test, adding more lines after line 74, you will have an inconsistent cleanup. IMO, the right place continues to be teardown.